### PR TITLE
Added Time Travel Reference

### PR DESCRIPTION
**Pull request by YiyunNi**

### CODE REVIEW

**Review by claireyuw — CHANGES_REQUESTED** [review]
Please decide the scope of this reference doc as a *user doc* and work with the engineer to see if we should omit certain concepts in this topic or add more reference docs, e.g. a knowhere introduction doc.

File: site/en/reference/timetravel_ref.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+Searches with filtering in knowhere is achieved by bitmap. Bitmap can be applied in the following three aspects:
+
+- Delete data
+- Timestamp
+- Attribute filtering
+
+When searching in segcore, it obtains a bitmap indicating if the timestamp meets the condition. Then, the segcore combines the timestamp bitmap with the other two types of bitmaps, data deletion bitmap and attribute filtering bitmap. Finally, a bitmap containing all deletion, attribute filtering, and timestamp information is generated.
+
+The search implementation for sealed segments and growing segments are slightly different. Because all CRUD operations within Milvus are executed in memory.  For sealed segments, you need to [load collection from disk to memory](search.md#Load-collection) before searching with Time Travel. However, for growing segments, you can search or use the Time Travel feature directly.
```
[the remark this answers is not in the document]
Comment (YiyunNi) [reply site/en/reference/timetravel_ref.md:31]: Yes

File: site/en/reference/timetravel_ref.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+Data engineers often need to roll back data to fix dirty data or bugs. Unlike traditional databases that use snapshots or retrain data to achieve data rollback, Milvus maintains a timeline for all data insert or delete operations. Therefore, users can specify the timestamp in a query to retrieve data at a specific point of time, which can significantly reduce maintenance costs.
+
+## Design Details
+
+When the proxy receives a data insert or delete request, it also gets a timestamp from root coord. Then, the proxy adds the timestamp as an additional field to the inserted or deleted data. All primary keys of the same insert or delete request have the same timestamp. The timestamp field is stored together with other data fields of a collection.
+
+When you load a collection to memory, all data in the collection and the timestamps are loaded into memory.
+
+During a search, if the search request received by the proxy contains the parameter, `travel_timestamp`, the value of this parameter will be passed to segcore, which filters the search results by timestamp.
```
[the remark this answers is not in the document]
Comment (czs007) [reply site/en/reference/timetravel_ref.md:19]: yes. @fishpenguin please introduce segcore briefly.

[the remark this answers is not in the document]
Comment (fishpenguin) [reply site/en/reference/timetravel_ref.md:19]: Segcore is the execution engine which supports concurrent insertion, deletion, query, index loading, monitoring and statistics of a segment data in memory.

File: site/en/reference/timetravel_ref.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+## Search implementation
+
+Searches with filtering in [knowhere](https://github.com/milvus-io/milvus/blob/master/docs/design_docs/knowhere_design.md) is achieved by bitmap. Bitmap can be applied in the following three aspects:
+
+- Delete data
+- Timestamp
+- Attribute filtering
+
+When searching in [segcore](https://github.com/milvus-io/milvus/tree/master/docs/design_docs/segcore), the execution engine which supports concurrent insertion, deletion, query, index loading, monitoring and statistics of a segment data in memory, it obtains a bitmap indicating if the timestamp meets the condition. Then, the segcore combines the timestamp bitmap with the other two types of bitmaps, data deletion bitmap and attribute filtering bitmap. Finally, a bitmap containing all deletion, attribute filtering, and timestamp information is generated.
```
[the remark this answers is not in the document]
Comment (claireyuw) [reply site/en/reference/timetravel_ref.md:29]: And what happens after this bitmap containing all information is generated?

**Review by claireyuw — CHANGES_REQUESTED** [review]

**Pull request merged**